crates/resolver/src/quic.rs

```diff
@@ -25,11 +26,12 @@ pub(crate) fn new_quic_stream(
     bind_addr: Option<SocketAddr>,
     dns_name: String,
     client_config: Option<TlsClientConfig>,
-) -> DnsExchangeConnect<QuicClientConnect, QuicClientStream, TokioTime> {
-    let client_config = client_config.map_or_else(
-        || CLIENT_CONFIG.clone(),
-        |TlsClientConfig(client_config)| client_config,
-    );
+) -> Result<DnsExchangeConnect<QuicClientConnect, QuicClientStream, TokioTime>, ResolveError> {
+    let client_config = if let Some(TlsClientConfig(client_config)) = client_config {
+        client_config
+    } else {
+        CLIENT_CONFIG.clone()?
+    };
```

Comment:
I could remove the returned `Result` here by building a `QuicClientConnect` `Future` that returns the error instead.

Reply:
The reason I ended up with a lot of those Connect types and the others, was to ensure that we had a consistent behavior to deal with in terms of Futures. I think it makes for a simpler API to return something like that Connect type. If others have a differing opinion I'm open to that.

Reply:
Done.